wire/core/ImageSizer.php

```diff
@@ -443,19 +446,31 @@ public function ___resize($targetWidth, $targetHeight = 0) {
 			if(self::checkMemoryForImage(array($targetWidth, $targetHeight, 3)) === false) {
 				throw new WireException(basename($source) . " - not enough memory to crop to the final image");
 			}
-
 			$thumb = imagecreatetruecolor($targetWidth, $targetHeight);
 			$this->prepareImageLayer($thumb, $image);
 			imagecopyresampled($thumb, $thumb2, 0, 0, $x1, $y1, $targetWidth, $targetHeight, $targetWidth, $targetHeight);
 			imagedestroy($thumb2);
 		}
 
+		if(isset($image) && is_resource($image)) @imagedestroy($image); // @horst
+		if(isset($thumb2) && is_resource($thumb2)) @imagedestroy($thumb2);
+		if(isset($image)) $image = null;
+		if(isset($thumb2)) $thumb2 = null;
+
 		// optionally apply sharpening to the final thumb
 		if($this->sharpening && $this->sharpening != 'none') { // @horst
 			if(IMAGETYPE_PNG != $this->imageType || ! $this->hasAlphaChannel()) {
-				// is needed for the USM sharpening function to calculate the best sharpening params
-				$this->usmValue = $this->calculateUSMfactor($targetWidth, $targetHeight);
-				$thumb = $this->imSharpen($thumb, $this->sharpening);
+				// calculate if there is enough memory available to apply the USM algorithm, if enabled
+				$this->useUSM = self::checkMemoryForImage(array(imagesx($thumb), imagesy($thumb), 3), array(imagesx($thumb), imagesy($thumb), 3)) === false ? false : $this->useUSM;
```

Comment:
Out of interest, shouldn't this check $this->useUSM first? If it's false in the first place, there seems to be no reason to perform a memory check, at least in this context :)